Test suite failed to run on with-jest example

# Examples bug report

## Example name
with-jest

## Describe the bug
Test suite failed to run

```
npm run test

> with-jest@1.0.0 test /Users/randalmaia/Code/next.js/examples/with-jest
> NODE_ENV=test jest

 FAIL  __tests__/index.test.js
  ● Test suite failed to run

    Plugin 0 specified in "/Users/randalmaia/Code/next.js/examples/with-jest/node_modules/next/babel.js" provided an invalid property of "default" (While processing preset: "/Users/randalmaia/Code/next.js/examples/with-jest/node_modules/next/babel.js")

      at Plugin.init (node_modules/jest-config/node_modules/babel-core/lib/transformation/plugin.js:131:13)
          at Array.map (<anonymous>)
          at Array.map (<anonymous>)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        1.445s
Ran all test suites.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! with-jest@1.0.0 test: `NODE_ENV=test jest`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the with-jest@1.0.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/randalmaia/.npm/_logs/2018-09-12T11_20_40_999Z-debug.log
```



## To Reproduce
- Clone the next.js repository 
```git clone git@github.com:zeit/next.js.git```
- Go to the example folder
``` cd next.js/examples/with-jest```
- Install the npm 
``` npm i ```
- Try run tests
```npm run test```

## Expected behavior
Should run the tests with success

## Screenshots
<img width="980" alt="screen shot 2018-09-12 at 8 13 08 am" src="https://user-images.githubusercontent.com/2751058/45421547-c224a880-b663-11e8-8962-c5e7411993e3.png">


## System information
 - OS: OSX
 - Version of Next.js: latest

## Additional context

I read this issue https://github.com/zeit/next.js/issues/4227 and tried the solutions without success

--- a/examples/with-jest/package.json
+++ b/examples/with-jest/package.json
@@ -3,12 +3,12 @@
   "version": "1.0.0",
   "dependencies": {
     "next": "latest",
-    "react": "16.4.2",
-    "react-dom": "16.4.2"
+    "react": "^16.5.0",
+    "react-dom": "^16.5.0"
   },
   "devDependencies": {
     "babel-core": "7.0.0-bridge.0",
-    "babel-jest": "23.4.2",
+    "babel-jest": "^23.6.0",
     "enzyme": "3.4.3",
     "enzyme-adapter-react-16": "1.2.0",
     "jest": "23.5.0",

